### ISSUE TITLE
Regarding the dataset iteration

### ISSUE THREAD

**Issue by Delicious-Bitter-Melon** [ByteDance-Seed/Bagel#211]
Thank you for open-sourcing the training code. I noticed that there are multiple IterableDataset instances within PackedDataset. I'd like to ask whether, during iteration, if one of the IterableDataset runs out of data and raises a StopIteration, this would cause the entire epoch to terminate early?

**Comment by Andy1621** [comment]
Hi, the code will not raise a StopIteration with `While True`.

**Comment by Delicious-Bitter-Melon** [comment]
> Hi, the code will not raise a StopIteration with `While True`.

Thanks. Got it.

**Comment by Delicious-Bitter-Melon** [comment]
Hi, I have one more question: does item['special_token_loss'] == 1 indicate that <eos> is included in the loss as the target token, or that it is used as a conditioning token and the predicted next token is the one included in the loss?
